### PR TITLE
Changes between project_version and dbt_version

### PR DESCRIPTION
**Pull request by McKnight-42**
changes made based on suggestions and continued discussion https://github.com/dbt-labs/dbt-database-adapter-scaffold/pull/31#discussion_r858652708

### CODE REVIEW

**Comment by McKnight-42** [comment]
other ideas regarding changes for this.
- deleted  https://github.com/dbt-labs/dbt-database-adapter-scaffold/blob/main/.github/workflows/version.yml#L37-L42 it would stop triggering every patch release but could this put us to far behind?
- possibly delete `dbt_version` all together and just use `project_version` even if it puts us a tad behind latest only upping to match minor and major releases.

**Comment by jtcohen6** [comment]
I'm very supportive of this! Want to hear what other folks think as well.

> even if it puts us a tad behind latest

I think this won't be an issue because of the `~=` compatibility operator. It's totally ok for the plugin version to be v1.1.0 even if `dbt-core` is already v1.1.2.

File: README.md [hunk 1]
```diff
@@ -9,6 +9,9 @@ dbt is the T in ELT. Organize, cleanse, denormalize, filter, rename, and pre-agg
 ## dbt-database-adapter-scaffold
 The `dbt-database-adapter-scaffold` package is a user-friendly interactive way to build out a new adapter.
 
+### Versioning
+In order to keep new adapters in a parity level with `dbt-core` when you generate your scaffold you will notice that the `project_version` might be something like `M.m.0` where M is equivalant to the latest major release of `dbt-core` and likewise for minor versioning. You can feel free to change these as you see fit. But would suggest that if you do keep the versioning as it while building out the adapter to note how new a software it is in the README.
```
Comment (kwigley) [inline README.md:13]:
```suggestion
`dbt-core` use semver to denote compatibility and intends to maintain compatibility for database adapters of the same major and minor version. (i.e. any patch version for a given major and minor version) This is a convention used by existing plugins that database adapter plugin developers can opt into during project setup. When generating a scaffold, the project version will include the same major and minor version of the latest stable version of `dbt-core`.
```

**Review by kwigley — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (McKnight-42) [reply README.md]: made suggested changes

**Pull request merged**